### PR TITLE
[WIP] black market

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
  https://github.com/tgstation/tgstation/pull/48120

issues that need fixing: 
 - the TGUI for the black market uplink's interface doesn't open

things to change after it works:
 - change some of the black market options available so nothing's overpowered or completely useless
 - make sure things cost more than they sell for at cargo (? maybe maybe not, since it's limited buying and flipping could be quirky)

## Why It's Good For The Game
  money but useful (sometimes)
someone help

## Changelog
:cl: TheChosenEvilOne
add: The black market, accessible from your local shitty hand-crafted black market uplink.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
this is a call for help

File: code/modules/cargo/blackmarket/blackmarket_items/consumables.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	stock_min = 5
+	stock_max = 35
+	price_min = 10
+	price_max = 60
+	availability_prob = 50
+
+/datum/blackmarket_item/consumable/pumpup
+	name = "Shoddy Stimulants"
+	desc = "Feel the energy inside each needle!"
+	/obj/item/reagent_containers/hypospray/medipen/stimpack
```
Comment (lolman360) [inline code/modules/cargo/blackmarket/blackmarket_items/consumables.dm]:
```suggestion
	item = /obj/item/reagent_containers/hypospray/medipen/stimpack
```

Comment (lolman360) [reply code/modules/cargo/blackmarket/blackmarket_items/consumables.dm]: do we even have pumpup

**Review by lolman360 — CHANGES_REQUESTED** [review]

File: code/modules/cargo/blackmarket/blackmarket_items/consumables.dm (hunk 1 above)
Comment (ghost) [reply code/modules/cargo/blackmarket/blackmarket_items/consumables.dm]: no we don't and i was too lazy to port it at the time might do later though (which is why i left the path name as pumpup)

File: code/modules/cargo/blackmarket/blackmarket_uplink.dm [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+				return
+			if(isnull(selected_item))
+				buying = FALSE
+				return
+			var/datum/blackmarket_market/market = SSblackmarket.markets[viewing_market]
+			market.purchase(selected_item, viewing_category, params["method"], src, usr)
+			buying = FALSE
+			selected_item = null
+
+/datum/crafting_recipe/blackmarket_uplink
```
Comment (ghost) [inline code/modules/cargo/blackmarket/blackmarket_uplink.dm]: i might boutta get retarded if this was the issue

Comment (ghost) [reply code/modules/cargo/blackmarket/blackmarket_uplink.dm]: i'm so fucking mad

**Comment by ghost** [comment]
by the way i've stopped working on this for now but i'll leave it open since kevinz told me lettern was porting things soon ™️ in hardsync 1.5 that might make this usable

**Comment by DeltaFire15** [comment]
Going to close this for now, feel free to ping me either here or in a dev-channel on discord when you want this reopened (aka, the stuff this PR needs is done and you feel like continuing work on it)

**Pull request closed without merging**